wrappers/csharp/Intel.RealSense/Device.cs

```diff
@@ -156,13 +156,16 @@ public SensorList Sensors
 
     public class AdvancedDevice : Device
     {
-        private const string AdvancedModeDisabledException = "Advanced mode has not been enabled";
-
         internal AdvancedDevice(IntPtr dev) : base(dev)
         {
 
         }
 
+        public static AdvancedDevice FromDevice(Device dev)
```

Comment:
Ah! this is what I was talking about before about using the is_extendable_to function...
This is the place to use the following code:
```csharp
object error;
if(NativeMethods.rs2_is_device_extendable_to(dev.m_instance, Extension.AdvancedMode, out error) <= 0)
{
    throw new ArgumentException("Device does not support AdvancedMode");
}
```

Reply:
That makes sense to me now :D I'll add the check.

Reply:
@zivsha But shouldn't it be 
```
if(NativeMethods.rs2_is_device_extendable_to(dev.m_instance, Extension.AdvancedMode, out error) == 0)
```
since the documentation states 

> * \return               Non-zero value iff the device can be extended to the given extension

Reply:
Probably you are right, I just complemented the `if (NativeMethods.rs2_is_frame_extendable_to(ptr, Extension.Points, out error) > 0)` in `FrameSet.cs`, looking at the code (that apparently I wrote), `==0` is the right way here.